code/modules/atmospherics/gasmixtures/gas_types.dm

```diff
@@ -31,6 +31,15 @@ GLOBAL_LIST_INIT(nonreactive_gases, typecacheof(list(/datum/gas/oxygen, /datum/g
 			return path
 	return ""
 
+/proc/gas_name2path(name)
+	var/list/meta_gas = GLOB.meta_gas_info
+	if(name in meta_gas)
+		return name
+	for(var/path in meta_gas)
+		if(meta_gas[path][META_GAS_NAME] == name)
+			return path
+	return ""
```

Comment:
I don't think this should be needed, see my above review. But if you're going to use it this if(name in meta_gas) shouldn't be here. You expect a name as input, not the typepath

Reply:
oh and this still has the upper/lower case problem